include/industry_standard/spdm.h

```diff
@@ -1214,6 +1224,66 @@ typedef struct {
     /*uint8_t                mel[portion_length];*/
 } spdm_measurement_extension_log_response_t;
 
+/* Key pair capabilities */
+#define SPDM_KEY_PAIR_CAP_GEN_KEY_CAP 0x00000001
+#define SPDM_KEY_PAIR_CAP_ERASABLE_CAP 0x00000002
+#define SPDM_KEY_PAIR_CAP_CERT_ASSOC_CAP 0x00000004
+#define SPDM_KEY_PAIR_CAP_KEY_USAGE_CAP 0x00000008
+#define SPDM_KEY_PAIR_CAP_ASYM_ALGO_CAP 0x00000010
+#define SPDM_KEY_PAIR_CAP_SHAREABLE_CAP 0x00000020
+#define SPDM_KEY_PAIR_CAP_MASK ( \
+        SPDM_KEY_PAIR_CAP_GEN_KEY_CAP | \
+        SPDM_KEY_PAIR_CAP_ERASABLE_CAP | \
+        SPDM_KEY_PAIR_CAP_CERT_ASSOC_CAP | \
+        SPDM_KEY_PAIR_CAP_KEY_USAGE_CAP | \
+        SPDM_KEY_PAIR_CAP_ASYM_ALGO_CAP | \
+        SPDM_KEY_PAIR_CAP_SHAREABLE_CAP)
+
+/* Key pair asym algorithm capabilities */
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA2048 0x00000001
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA3072 0x00000002
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA4096 0x00000004
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_ECC256 0x00000008
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_ECC384 0x00000010
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_ECC521 0x00000020
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_SM2 0x00000040
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_ED25519 0x00000080
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_ED448 0x00000100
+#define SPDM_KEY_PAIR_ASYM_ALGO_CAP_MASK ( \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA2048 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA3072 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA4096 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_ECC256 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_ECC384 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_ECC521 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_SM2 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_ED25519 | \
+        SPDM_KEY_PAIR_ASYM_ALGO_CAP_ED448)
+
+#define SPDM_MAX_PUBLIC_KEY_INFO_LEN 32
```

Comment:
This is `65535`.

Reply:
In SPDM spec, the public key info is: 
```
The format of this field shall be the DER encoding of the AlgorithmIdentifier structure
in an X.509 v3 certificate.
See the "4.1.2.7. Subject Public Key Info" clauses in RFC 5280 for additional details.
```

In RFC5280 spec, the AlgorithmIdentifier structure is:
![image](https://github.com/user-attachments/assets/07e9046d-f8ad-40ad-b874-c5c7dc525f69)

 * The RSA public key info len is 15.
 * The ecp256 public key info len is 21.
 * The ecp384 public key info len is 18.
 * The ecp521 public key info len is 18.
 * The sm2 public key info len is 21.
 * The ed25519 public key info len is 7.
 * The ed448 public key info len is 7.

So the 32 is enough for SPDM_MAX_PUBLIC_KEY_INFO_LEN.

Reply:
>So the 32 is enough for SPDM_MAX_PUBLIC_KEY_INFO_LEN.

That's fine for a `LIBSPDM_` macro but this macro is for the SPDM specification and `PublicKeyInfoLen` is two bytes. We do the same thing for https://github.com/DMTF/libspdm/blob/69278feba36eb1fc661bd21d995ee309f986283a/include/industry_standard/spdm.h#L567-L568 where a certificate chain is typically only a few KiBs.

Reply:
Thanks. I have changed the SPDM_MAX_PUBLIC_KEY_INFO_LEN to 32.